Sync `html/dom` from WPT upstream (exclude `render-blocking`)

Ahmad-S792 opened the pull request:
#### 889829b14625f9dab90caba0ea5d08b840e81258
<pre>
Sync `html/dom` from WPT upstream (exclude `render-blocking`)

<a href="https://bugs.webkit.org/show_bug.cgi?id=271366">https://bugs.webkit.org/show_bug.cgi?id=271366</a>

Reviewed by Anne van Kesteren.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/847ab6342ca01621db8bd7f38f8e31d7686dd95e">https://github.com/web-platform-tests/wpt/commit/847ab6342ca01621db8bd7f38f8e31d7686dd95e</a>

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/aria-attribute-reflection-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/aria-attribute-reflection.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/aria-element-reflection-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/aria-element-reflection.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/documents/resource-metadata-management/document-lastModified-01-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/documents/resource-metadata-management/document-lastModified-01.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/dir-assorted.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/dir-assorted.window.js:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/dir-shadow-41-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/dir-shadow-41-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/dir_auto-N-EN-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/dir_auto-N-EN-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/dir_auto-N-EN.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/the-anchor-attribute-003.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/the-innertext-and-outertext-properties/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/idlharness-shadowrealm.window.js:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/idlharness.https.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/idlharness.https_include=HTML._-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/idlharness.worker.js:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-embedded-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-forms-weekmonth-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-grouping-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-metadata-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-obsolete-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-sections-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-tabular-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-text-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection.js:
(ReflectionTests.parseFloat):
(ReflectionTests.typeMap.string_appeared_here.valueOf):
(ReflectionTests.typeMap.string_appeared_here.string_appeared_here):
(ReflectionTests.typeMap.string_appeared_here.toString):
(ReflectionTests.reflects):
* LayoutTests/imported/w3c/web-platform-tests/html/dom/w3c-import.log:
* LayoutTests/platform/mac-wk2/imported/w3c/web-platform-tests/html/dom/reflection-forms-expected.txt:
* LayoutTests/platform/mac-wk2/imported/w3c/web-platform-tests/html/dom/reflection-misc-expected.txt:

&gt; Platform Specific Rebaselines:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/dom/idlharness.https_include=HTML._-expected.txt: Rebaselined
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/dom/reflection-forms-expected.txt&quot; Ditto
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/dom/reflection-misc-expected.txt: Ditto
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/dom/idlharness.https_include=HTML._-expected.txt: Ditto
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/dom/reflection-forms-expected.txt: Ditto
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/dom/reflection-forms-weekmonth-expected.txt: Ditto
* LayoutTests/imported/w3c/web-platform-tests/html/dom/idlharness.https_exclude=(Document_Window_HTML._)-expected.txt: Ditto
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/dom/idlharness.https_include=HTML._-expected.txt: Ditto
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/dom/reflection-embedded-expected.txt: Ditto
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/dom/reflection-misc-expected.txt: Added new platform expectation
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/dom/reflection-forms-expected.txt: Rebaselined
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/dom/reflection-forms-weekmonth-expected.txt: Ditto
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/dom/reflection-misc-expected.txt: New Test Expectation
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/dom/reflection-misc-expected.txt: Ditto
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-misc-expected.txt: Rebaselined
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/html/dom/reflection-forms-expected.txt: Rebaselined
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/dom/reflection-embedded-expected.txt: Ditto
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/dom/reflection-forms-expected.txt: Ditto
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/dom/reflection-misc-expected.txt: Ditto

Canonical link: <a href="https://commits.webkit.org/276867@main">https://commits.webkit.org/276867@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5f7f94d528f22cb090814fcd5f50dfe286aa3ea5

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/46028 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/25157 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/48613 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/48699 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/42068 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/29470 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/22576 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/37634 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/46606 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/22243 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/39708 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/18825 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/19640 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/40804 "Passed tests") | [✅ 🛠 wpe-skia](https://ews-build.webkit.org/#/builders/52/builds/4072 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/42336 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/41155 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/50485 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/21024 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/17522 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/44793 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/22324 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/39866 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/43684 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/10191 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/22683 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/22018 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

annevk reviewed (approved):
r+ modulo bots.

Nit: it's `html/dom` (forward slash).

Commit:
Sync `html/dom` from WPT upstream (exclude `render-blocking`)

https://bugs.webkit.org/show_bug.cgi?id=271366

Reviewed by Anne van Kesteren.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/847ab6342ca01621db8bd7f38f8e31d7686dd95e

* LayoutTests/imported/w3c/resources/resource-files.json:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/aria-attribute-reflection-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/aria-attribute-reflection.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/aria-element-reflection-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/aria-element-reflection.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/documents/resource-metadata-management/document-lastModified-01-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/documents/resource-metadata-management/document-lastModified-01.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/dir-assorted.window-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/dir-assorted.window.js:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/dir-shadow-41-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/dir-shadow-41-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/dir_auto-N-EN-expected.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/dir_auto-N-EN-ref.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/dir_auto-N-EN.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/the-anchor-attribute-003.tentative.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/global-attributes/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/elements/the-innertext-and-outertext-properties/w3c-import.log:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/idlharness-shadowrealm.window.js:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/idlharness.https.html:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/idlharness.https_include=HTML._-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/idlharness.worker.js:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-embedded-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-forms-weekmonth-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-grouping-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-metadata-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-obsolete-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-sections-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-tabular-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-text-expected.txt:
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection.js:
(ReflectionTests.parseFloat):
(ReflectionTests.typeMap.string_appeared_here.valueOf):
(ReflectionTests.typeMap.string_appeared_here.string_appeared_here):
(ReflectionTests.typeMap.string_appeared_here.toString):
(ReflectionTests.reflects):
* LayoutTests/imported/w3c/web-platform-tests/html/dom/w3c-import.log:
* LayoutTests/platform/mac-wk2/imported/w3c/web-platform-tests/html/dom/reflection-forms-expected.txt:
* LayoutTests/platform/mac-wk2/imported/w3c/web-platform-tests/html/dom/reflection-misc-expected.txt:

> Platform Specific Rebaselines:
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/dom/idlharness.https_include=HTML._-expected.txt: Rebaselined
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/dom/reflection-forms-expected.txt" Ditto
* LayoutTests/platform/ios-wk2/imported/w3c/web-platform-tests/html/dom/reflection-misc-expected.txt: Ditto
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/dom/idlharness.https_include=HTML._-expected.txt: Ditto
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/dom/reflection-forms-expected.txt: Ditto
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/dom/reflection-forms-weekmonth-expected.txt: Ditto
* LayoutTests/imported/w3c/web-platform-tests/html/dom/idlharness.https_exclude=(Document_Window_HTML._)-expected.txt: Ditto
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/dom/idlharness.https_include=HTML._-expected.txt: Ditto
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/dom/reflection-embedded-expected.txt: Ditto
* LayoutTests/platform/glib/imported/w3c/web-platform-tests/html/dom/reflection-misc-expected.txt: Added new platform expectation
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/dom/reflection-forms-expected.txt: Rebaselined
* LayoutTests/platform/mac-wk1/imported/w3c/web-platform-tests/html/dom/reflection-forms-weekmonth-expected.txt: Ditto
* LayoutTests/platform/gtk/imported/w3c/web-platform-tests/html/dom/reflection-misc-expected.txt: New Test Expectation
* LayoutTests/platform/wpe/imported/w3c/web-platform-tests/html/dom/reflection-misc-expected.txt: Ditto
* LayoutTests/imported/w3c/web-platform-tests/html/dom/reflection-misc-expected.txt: Rebaselined
* LayoutTests/platform/mac/imported/w3c/web-platform-tests/html/dom/reflection-forms-expected.txt: Rebaselined
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/dom/reflection-embedded-expected.txt: Ditto
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/dom/reflection-forms-expected.txt: Ditto
* LayoutTests/platform/ios/imported/w3c/web-platform-tests/html/dom/reflection-misc-expected.txt: Ditto

Canonical link: https://commits.webkit.org/276867@main
